python/lsst/display/matplotlib/matplotlib.py

```diff
@@ -275,22 +319,41 @@ def _i_mtv(self, data, wcs, title, isMask):
 
                 colors[i + 1][3] = alpha
 
-            dataArr = maskArr
             cmap = mpColors.ListedColormap(colors)
             norm = mpColors.NoNorm()
         else:
-            cmap = pyplot.cm.gray
+            cmap = self._image_colormap
             norm = self._normalize
 
         ax = self._figure.gca()
         bbox = data.getBBox()
-        mappable = ax.imshow(dataArr, origin='lower', interpolation='nearest',
-                             extent=(bbox.getBeginX() - 0.5, bbox.getEndX() - 0.5,
-                                     bbox.getBeginY() - 0.5, bbox.getEndY() - 0.5),
-                             cmap=cmap, norm=norm)
+        extent = (bbox.getBeginX() - 0.5, bbox.getEndX() - 0.5,
+                  bbox.getBeginY() - 0.5, bbox.getEndY() - 0.5)
+
+        with pyplot.rc_context(dict(interactive=False)):
+            if isMask:
+                for i, p in reversed(list(enumerate(planeList))):
+                    if colors[i + 1][3] == 0:
+                        continue
+
+                    bitIsSet = (dataArr & (1 << p)) != 0
+                    if bitIsSet.sum() == 0:
+                        continue
 
-        if not isMask:
-            self._mappable = mappable
+                    maskArr[bitIsSet] = i + 1  # + 1 as we set colorNames[0] to black
+
+                    if not self._fastMaskDisplay:  # we draw each bitplane separately
+                        ax.imshow(maskArr, origin='lower', interpolation='nearest',
+                                  extent=extent, cmap=cmap, norm=norm)
+                        maskArr[:] = 0
+
+                if self._fastMaskDisplay:  # we only draw the lowest bitplane
```

Comment:
This comment should say 'we only draw the lowest bitplane set for each pixel'.

`maskArr[bitIsSet] = i + 1` accumulates for each item in `planeList`.  The `maskArr` is only reset if `not self._fastMaskDisplay`.